src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsServiceCollectionExtensions.cs

```diff
@@ -159,5 +161,54 @@ public static IWebJobsBuilder AddWebJobs(this IServiceCollection services, Actio
 
             return builder;
         }
+
+        /// <summary>
+        /// Adds the WebJobs scale services to the provided <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        public static IWebJobsBuilder AddWebJobsScale(this IServiceCollection services, Action<ScaleOptions> configure)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.AddCommonScaleServices();
+            services.TryAddSingleton<IPrimaryHostStateProvider>(new PrimaryHostStateProvider() { IsPrimary = true });
+
+            if (configure != null)
+            {
+                services.Configure(configure);
+            }
+
+            // Options logging
+            services.AddTransient(typeof(OptionsFactory<>));
```

Comment:
Why are we adding these options services as part of the scale logic? In the context of Functions, this is already handled, but for WebJobs, this shouldn't be tied to this logic.

Reply:
Yeah, it's not really. AddWebJobsScale configures a host as a ScaleHost, and we want the options logging stuff as well to log ScaleOptions. So it's common registrations that we want to do for normal WebJobs hosts as well as scaler hosts. Could be factored into a shared method like I did for AddCommonScaleServices.

Reply:
can you call `services.AddOptions()` instead? That conditionally registers everything you'd need for Options -- and if they change something in future versions, it'd continue to work.

https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Options/src/OptionsServiceCollectionExtensions.cs#L22

Reply:
This was copy pasted from [here](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsServiceCollectionExtensions.cs#L119).

Reply:
Bunch of the tests are broken after the change:
https://github.com/Azure/azure-webjobs-sdk/pull/2954/checks?check_run_id=11964665844

Do we want to fix them all revert `services.AddOptions()`?